app/controllers/phases_controller.rb

```diff
@@ -19,4 +36,42 @@ def set_phase
     @phase = Phase.where(challenge: current_user.challenge_manager_challenges).find(params[:id])
     @challenge = @phase.challenge
   end
+
+  def apply_filters
+    apply_status_filter if params[:status]
+    apply_eligibility_filters
+  end
+
+  def apply_status_filter
+    @submissions = case params[:status]
+                   when 'not_started' then @not_started
+                   when 'in_progress' then @in_progress
+                   when 'completed'   then @completed
+                   when 'recused'     then filter_recused_submissions
+                   else @submissions
+                   end
+  end
+
+  def filter_recused_submissions
+    @submissions.joins(:evaluator_submission_assignments).
+      where(evaluator_submission_assignments: { status: :recused })
+  end
+
+  def apply_eligibility_filters
+    @submissions = @submissions.select(&:eligible_for_evaluation?) if params[:eligible_for_evaluation] == 'true'
+    @submissions = @submissions.select(&:selected_to_advance?) if params[:selected_to_advance] == 'true'
```

Comment:
for performance it might be better to apply this filter to the SQL query rather than in-memory. actually these could be useful scopes on the model.
```suggestion
    @submissions = @submissions.where(judging_status: [:selected, :winner]) if params[:eligible_for_evaluation] == 'true'
    @submissions = @submissions.where(judging_status: :selected) if params[:selected_to_advance] == 'true'
```